api_client/python/timesketch_api_client/sigma.py

```diff
@@ -31,67 +32,267 @@ class Sigma(resource.BaseResource):
     """
 
 
-    def __init__(self, rule_uuid, api, es_query=None,
-                 file_name=None, title=None, description=None,
-                 file_relpath=None):
+    def __init__(self, api):
         """Initializes the Sigma object.
 
         Args:
-            rule_uuid: Id of the sigma rule.
             api: An instance of TimesketchApi object.
-            es_query: Elastic Search query of the rule
-            file_name: File name of the rule
-            title: Title of the rule
-            description: Description of the rule
-            file_relpath: path of the file relative to the config value
 
         """
-        self.rule_uuid = rule_uuid
-        self._description = description
-        self._es_query = es_query
-        self._file_name = file_name
-        self._title = title
-        self._file_relpath = file_relpath
-        self._resource_uri = f'sigma/{self.rule_uuid}'
+        self.api = api
+        self._author = None
+        self._date = None
+        self._description = None
+        self._detection = None
+        self._es_query = None
+        self._falsepositives = None
+        self._file_name = None
+        self._file_relpath = None
+        self._id = None
+        self._level = None
+        self._logsource = None
+        self._modified = None
+        self._rule_uuid = None
+        self._references = None
+        self._resource_uri = 'sigma/' # TODO: clarify: is that okay?
+        self._title = None
+
         super().__init__(
             api=api, resource_uri=self._resource_uri)
 
     @property
     def es_query(self):
         """Returns the elastic search query."""
-        sigma_data = self.data
+        if self._es_query:
+            return self._es_query
 
-        if not sigma_data:
-            return ''
+        if not self.data or self.data is None:
+            self.lazyload_data(refresh_cache=True)
```

Comment:
`self.data` calls `self.lazyload_data`, since that is set by the base resource....